### ISSUE TITLE
manual build with -DDLIB_NO_GUI_SUPPORT=0 giving weird bug

### ISSUE THREAD

**Issue by nateGeorge** [menpo/conda-dlib#10]
For some reason when I manually build, it's giving a weird bug.  

```
Boost.Python.ArgumentError: Python argument types in
    simple_object_detector.save(simple_object_detector, str)
did not match C++ signature:
    save(dlib::simple_object_detector_py, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > detector_output_filename)
```

I'm not sure if it's something about my local env variables or what.  When I do

`conda install -c menpo dlib=18.18`

It works fine, but I want to have the GUI enabled, so I'm trying to do a custom build.

**Comment by patricksnape** [comment]
What operating system?

**Comment by nateGeorge** [comment]
Ubuntu 16.04 LTS 64bit.  The error I posted is from trying to save a detector object.

**Comment by patricksnape** [comment]
Right, so that is likely due to the upgrade to GCC 5.x. It may have something to do with the slightly old build of Boost that I am using here. Likely Boost > 1.59 is required for C++11 builds.

**Comment by nateGeorge** [comment]
Hmm, so I should try to build with an older boost somehow?  It says it is installing boost 1.59 when it is installing dlib.  I also installed 1.61 to see if it helped, but I don't know which boost it's actually using.

**Comment by patricksnape** [comment]
I would build with a newer boost. I think conda-forge has a newer boost. Maybe if you try adding conda-forge to your `.condarc` channels list and then updating the boost in the recipe to 1.61?

**Comment by nateGeorge** [comment]
Alright, I'll see if I can figure that out.  Thanks!

**Comment by patricksnape** [comment]
@nateGeorge Did you ever get this working?

**Comment by nateGeorge** [comment]
Nope, I never did.  Just using python2 with dlib for the GUI.
